src/core/dev/net_device_table_mgr.cpp

```diff
@@ -264,19 +264,32 @@ void net_device_table_mgr::update_tbl()
                 if ((int)get_max_mtu() < p_net_device_val->get_mtu()) {
                     set_max_mtu(p_net_device_val->get_mtu());
                 }
+                auto handle_ip_insertion = [p_net_device_val](const std::unique_ptr<ip_data> &ip,
+                                                              auto &net_device_map,
+                                                              sa_family_t family) {
+                    auto it = net_device_map.find(ip->local_addr);
+                    if (it == net_device_map.end() ||
+                        p_net_device_val->get_state() != net_device_val::DOWN) {
+                        net_device_map[ip->local_addr] = p_net_device_val;
+                    }
+                    if (it != net_device_map.end()) {
+                        vlog_printf(VLOG_WARNING,
+                                    "IP ADDRESS %s ASSIGNED TO MULTIPLE INTERFACES NOT SUPPORTED\n",
+                                    ip->local_addr.to_str(family).c_str());
+                    }
```

Comment:
This approach can print multiple extra messages (per problematic interface). This is OK since this is unlikely scenario. We still can consider choosing among the following options:
1. Print a message per problematic interface as is. I'd prefer to move the if block at the top of the lambda not to think about iterator invalidation (the previous block manipulates with the collection).
2. Remember the fact that there is at least one duplicate IP and print a single common warning message at the end.
3. Create a map on stack with key=IP value=net_device_val and print a single message per IP at the end with mentioning which interface will be used for each IP.

Also rewrite the warning message:
 * Don't use upper case
 * Write more "light" message, because XLIO will work as expected in multiple scenarios even with duplicate IP addresses, and the current message makes impression that XLIO won't work at all. Use something like "Duplicate IP address %s is detected on multiple interfaces. XLIO will work with a single interface".